datatypes/src/raster/operations/map_elements.rs

```diff
@@ -0,0 +1,364 @@
+use crate::raster::{
+    data_type::DefaultNoDataValue, EmptyGrid, Grid, GridOrEmpty, GridSize, NoDataValue,
+    RasterTile2D,
+};
+use rayon::iter::{IndexedParallelIterator, IntoParallelRefIterator, ParallelIterator};
+
+pub trait MapElements<In, Out, F: Fn(In) -> Option<Out>> {
+    type Output;
+    fn map_elements(self, map_fn: F, out_no_data: Option<Out>) -> Self::Output;
+}
+
+pub trait MapElementsParallel<In, Out, F: Fn(In) -> Option<Out>> {
+    type Output;
+    fn map_elements_parallel(self, map_fn: F, out_no_data: Option<Out>) -> Self::Output;
+}
+
+impl<In, Out, F, G> MapElements<In, Out, F> for Grid<G, In>
+where
+    In: 'static + Copy + PartialEq,
+    Out: 'static + Copy + DefaultNoDataValue,
+    G: GridSize + Clone,
+    F: Fn(In) -> Option<Out>,
+{
+    type Output = Grid<G, Out>;
+    fn map_elements(self, map_fn: F, out_no_data: Option<Out>) -> Self::Output {
+        let out_no_data = out_no_data.unwrap_or(Out::DEFAULT_NO_DATA_VALUE);
+        let shape = self.shape.clone();
+        let data = self
+            .data
+            .iter()
+            .map(|x| {
+                if self.is_no_data(*x) {
+                    out_no_data
+                } else {
+                    map_fn(*x).unwrap_or(out_no_data)
```

Comment:
is this always meaningful or do we need a option (parameter) to return an error?

Reply:
i guess there might be a case where we need to return errors but we could add that later